code/parachain/frame/composable-traits/src/staking/lock.rs

```diff
@@ -60,19 +62,48 @@ impl<MaxDurationPresets: Get<u32>> DurationMultipliers<MaxDurationPresets> {
 			DurationMultipliers::Presets(presets) => presets.len() > 0,
 		}
 	}
+
+	/// Get the maximum stake duration allowed by this duration configuration. Returns [`None`] if
+	/// the stake durations are unbounded.
+	// TODO(benluelo): We need to verify that the duration presets are valid (i.e at least one
+	// duration provided)
+	pub fn max_duration(&self) -> Option<u64> {
+		match self {
+			DurationMultipliers::Presets(presets) =>
+				presets.last_key_value().map(|(k, _)| k).copied(),
```

Comment:
So we asssume last key is the max?

Reply:
Yeah, that's how BTreeMaps work: https://doc.rust-lang.org/stable/std/collections/struct.BTreeMap.html#method.last_key_value